crates/shielded_token/src/conversion.rs

```diff
@@ -155,17 +177,21 @@ where
         last_locked_dec,
     );
 
+    // total locked amount in the Shielded pool
+    let rewardable_tokens_in_masp =
+        get_masp_dated_balance::<S, TransToken>(storage, token)?;
+
     // inflation-per-token = inflation / locked tokens = n/PRECISION
     // ∴ n = (inflation * PRECISION) / locked tokens
     // Since we must put the notes in a compatible format with the
     // note format, we must make the inflation amount discrete.
-    let noterized_inflation = if total_tokens_in_masp.is_zero() {
```

Comment:
Do we still need to read the `total_tokens_in_masp` variable in the prelude of this function?

Reply:
I believe so. `total_tokens_in_masp` is still used in the inflation computation at https://github.com/anoma/namada/blob/cb6794c91f6aa4de2ae94232bd6cf744288a89e9/crates/shielded_token/src/conversion.rs#L133 . More broadly, the purpose of the shielded rewards system is to incentivise end users to lock their tokens until the target amount is reached, and both dated and undated amounts count as locked because they both contribute to the anonymity set.